<!--
  Standard launch configuration for an app manager in standalone mode.
  i.e. no android, no multimaster concert. You should be able to play
  the basic apps (chirp and talker etc with this).
 -->
<launch>
  <!-- ******************************* Arguments ******************************* -->
  <arg name="robot_name" default="cybernetic_pirate"/>
  <arg name="robot_type" default="turtlebot"/>
  <arg name="app_lists" default="$(find rocon_apps)/apps/rocon.rapps"/> <!-- semi colon separated list --> 
  <arg name="gateway_watch_loop_period" default="10"/> <!-- Polling period for multimaster advertising/flipping -->
  <arg name="hub_whitelist" default=""/> <!-- semi-colon separated hub names/regex patterns -->

  <!-- ******************************* Zeroconf ******************************** -->
  <node ns="zeroconf" pkg="zeroconf_avahi" type="zeroconf" name="zeroconf"/>

  <!-- ******************************** Gateway ******************************** -->
  <include file="$(find rocon_app_manager)/launch/includes/_app_manager_gateway.xml">
    <arg name="gateway_name" value="$(arg robot_name)" />
    <arg name="gateway_watch_loop_period" value="$(arg gateway_watch_loop_period)" />
    <arg name="hub_whitelist" value="$(arg hub_whitelist)" />
  </include>

  <!-- ****************************** App Manager ****************************** -->
  <include file="$(find rocon_app_manager)/launch/includes/_app_manager.xml">
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="robot_type" value="$(arg robot_type)" />
    <arg name="app_lists" value="$(arg app_lists)" />
  </include>
</launch>
